From ebe9144fdfdc8154661b93322cec4a1c43aa75ff Mon Sep 17 00:00:00 2001
From: alperak <alperyasinak1@gmail.com>
Date: Tue, 1 Oct 2024 19:45:38 +0300
Subject: [PATCH] Fix invalid conversion from int to CURLoption

Options.hpp:281:74: error: invalid conversion from 'int' to 'CURLoption' [-fpermissive]
  281 |         typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
      |                                                                          ^
      |                                                                          |
      |                                                                          int

Fix invalid conversion by adding explicit cast to CURLoption

Upstream-Status: Pending

Signed-off-by: alperak <alperyasinak1@gmail.com>
---
 include/curlpp/Options.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/curlpp/Options.hpp b/include/curlpp/Options.hpp
index c705c9d..7c9a501 100644
--- a/include/curlpp/Options.hpp
+++ b/include/curlpp/Options.hpp
@@ -278,7 +278,7 @@ namespace options
 	typedef curlpp::OptionTrait<long, CURLOPT_LOW_SPEED_LIMIT> LowSpeedLimit;
 	typedef curlpp::OptionTrait<long, CURLOPT_LOW_SPEED_TIME> LowSpeedTime;
 	typedef curlpp::OptionTrait<long, CURLOPT_MAXCONNECTS> MaxConnects;
-	typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
+	typedef curlpp::OptionTrait<curl_closepolicy, (CURLoption)CURLOPT_CLOSEPOLICY> ClosePolicy;
 	typedef curlpp::OptionTrait<bool, CURLOPT_FRESH_CONNECT> FreshConnect;
 	typedef curlpp::OptionTrait<bool, CURLOPT_FORBID_REUSE> ForbidReuse;
 	typedef curlpp::OptionTrait<long, CURLOPT_CONNECTTIMEOUT> ConnectTimeout;
-- 
2.25.1

